src/services/puzzlebang/puzzlebang-router.ts

```diff
@@ -11,25 +11,38 @@ puzzlebangRouter.post(
     "/",
     RoleChecker([Role.Enum.PUZZLEBANG]),
     async (req, res) => {
-        const requestInfo = PuzzlebangCompleteRequestValidator.parse(req.body);
-
-        const attendeeData = await Database.ATTENDEE.findOneAndUpdate(
-            { email: requestInfo.email },
-            {
-                $addToSet: { puzzlesCompleted: requestInfo.puzzleId },
-                $inc: { points: 2 },
-            },
-            { new: false }
-        );
-
-        if (!attendeeData) {
+        const { userId, puzzleId } = PuzzlebangCompleteRequestValidator.parse(req.body);
+        
+        const { data: attendeeData, error } = await SupabaseDB.ATTENDEES
+            .select("puzzles_completed, points")
+            .eq("user_id", userId)
+            .single();
+
+        if (error || !attendeeData) {
             return res.sendStatus(StatusCodes.NOT_FOUND);
```

Comment:
If it's an error, could we return a 500 error, preferably along with the error message?